### PR TITLE
Introduce "withdraw all" functionality

### LINKED ISSUE #32: Wallet: withdraw all function

### PR DESCRIPTION
**Pull request by thomaseizinger**
Fixes #32.

### CODE REVIEW

File: waves/wallet/src/wallet.rs [hunk 1]
```diff
@@ -210,6 +221,258 @@ pub async fn current<'n, 'w>(
...
+        .group_by(|(_, _, unblinded)| unblinded.asset);
+
+    // prepare the data exactly as we need it to create the transaction
+    let txouts_grouped_by_asset = (&txouts_grouped_by_asset)
+        .into_iter()
+        .map(|(asset, txouts)| {
+            let txouts = txouts.collect::<Vec<_>>();
+
+            // calculate the total amount we want to spend for this asset
+            // if this is the native asset, subtract the fee
```
Comment (luckysori) [inline waves/wallet/src/wallet.rs:295]: This reminds me that a user can't withdraw without sufficient L-BTC.

Comment (thomaseizinger) [reply waves/wallet/src/wallet.rs:295]: Yes, for now the UI could just call this function and display the error message? That would say that you need L-BTC.

We can also add an additional check in the UI!

File: waves/wallet/src/wallet.rs [hunk 2]
```diff
@@ -281,6 +544,29 @@ where
         .await
 }
 
+/// Estimate the virtual size of a transaction based on the number of inputs and outputs.
```
Comment (luckysori) [inline waves/wallet/src/wallet.rs:547]: :abacus:

File: waves/wallet/src/wallet.rs (hunk 1 above)
Comment (luckysori) [reply waves/wallet/src/wallet.rs:295]: I see that the error is handled below.

File: waves/wallet/src/wallet.rs [hunk 3]
```diff
@@ -210,6 +221,258 @@ pub async fn current<'n, 'w>(
...
+                asset,
+                total_input
+            );
+
+            (asset, txins, txout_inputs, to_spend)
+        })
+        .collect::<Vec<_>>();
+
+    // build transaction from grouped txouts
+    let mut transaction = match txouts_grouped_by_asset.as_slice() {
```
Comment (luckysori) [inline waves/wallet/src/wallet.rs:352]: I love using subslice patterns!

**Review by luckysori — APPROVED** [review]
That `withdraw_everything_to` function looks crazy :eyes:

File: waves/wallet/src/wallet.rs (hunk 3 above)
Comment (thomaseizinger) [reply waves/wallet/src/wallet.rs:352]: They are great right? Loving it too!

**Pull request merged**